src/Data/Lens/At.purs

```diff
@@ -0,0 +1,45 @@
+module Data.Lens.At
+  ( At
+  , at
+  ) where
+
+import Prelude
+
+import Data.Identity (runIdentity, Identity(..))
+import Data.Map as M
+import Data.Maybe (Maybe(..), maybe)
+import Data.Set as S
+import Data.StrMap as SM
+import Data.Tuple
+
+import Data.Lens (LensP(), lens)
+import Data.Lens.Index (Index)
+
+class (Index m a b) <= At m a b where
```

Comment:
I know this comes from `purescript-index` but let's discuss the API a bit. Why not

```purescript
class Index m k where
  ix :: forall a. k -> TraversalP (m a) a 

class (Index m k) <= At m k where
  at :: forall a. k -> LensP (m a) (Maybe a)
```

Namely, parameterize over the type constructor to drop one type argument, and ditch `IndexKey` etc. which don't get used anyway.